### PR TITLE
1.14 release notes

### PR DESCRIPTION
**Pull request by brson**
[Rendered](https://github.com/brson/rust/blob/relnotes/RELEASES.md).

Is there anything I'm missing? Were there any library stabilizations?

The most interesting thing this time seems to be rustup, then compile time opts, `..` matches, cargo exposing `cfg` values to build scripts. This release also has wasm support, but it's pretty rough. Might be worth hyping up all the same. ARM MUSL targets.

There are a large number of compatibility notes. Might be worth talking about that.

Relevant prlo links: goo.gl/PGz2Ds, goo.gl/KV23Qv, goo.gl/g7ku6D.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,203 @@
...
+* [Implement `RefUnwindSafe` for atomic types][37178]
+* [Specialize `Vec::extend` to `Vec::extend_from_slice`][37094]
+* [Avoid allocations in `Decoder::read_str`][37064]
+* [`io::Error` implements `From<io::ErrorKind>`][37037]
+* [Impl `Debug` for raw pointers to unsized data][36880]
+* [Don't reuse `HashMap` random seeds][37470]
+* [The internal memory layout of `HashMap` is more cache-friendly, for
+  significant improvements in some operations][36692]
+* [`HashMap` uses less memory on 32-bit architectures][36595]
+* [Impl `Add<{str, Cow<str>}>` for `Cow<str>`][36595]
```
Comment (Arnavion) [inline RELEASES.md]: This should be 36430?

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,203 @@
...
+  `armv7-unknown-linux-musleabihf` targets via `rustup target add`.
+* rustup no longer installs documentation by default. Run `rustup
+  component add rust-docs` to install.
+* [Fix line stepping in debugger][37310]
+* [Enable line number debuginfo in releases][37280]
+
+Misc
+----
+
+* [Disable jemalloc on aarch64/powerpc][37392]
```
Comment (xen0n) [inline RELEASES.md]: MIPS is also touched by this PR, but the PR title didn't reflect that.

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,203 @@
...
+* [Ignore `panic` configuration for test/bench profiles][cargo/3175]
+
+Tooling
+-------
+
+* [rustup is the recommended Rust installation method][1.14rustup]
+* This release includes [experimental support for WebAssembly][1.14wasm]
+* This release includes builds for ARM Linux running MUSL libc. Add the
+  `arm-unknown-linux-musleabi`, `arm-unknown-linux-musleabihf`, and
+  `armv7-unknown-linux-musleabihf` targets via `rustup target add`.
```
Comment (xen0n) [inline RELEASES.md]: Might also mention the addition of MIPS/MIPS64/PPC/s390x host builds since they didn't appear in 1.13.0 stable.

Comment (brson) [inline RELEASES.md]: Good idea. I only with the rustup mips build worked...

**Comment by alexcrichton** [comment]
Looks great to me! r=me with comments addressed

**Comment by brson** [comment]
All comments so far addressed. I really beefed up the bullets on platform support. It seems like rustup + platform support is the big story here.

**Comment by brson** [comment]
I'm going to backport this revision to 1.14.

**Comment by brson** [comment]
I'm not going to bother backporting after all.

**Pull request merged**